tests/integration/pilot/gateway_test.go

```diff
@@ -181,6 +182,45 @@ spec:
 	})
 }
 
+func ManagedGatewayShortNameTest(t framework.TestContext) {
+	t.ConfigIstio().YAML(apps.Namespace.Name(), `apiVersion: gateway.networking.k8s.io/v1beta1
+kind: Gateway
+metadata:
+  name: gateway
+spec:
+  gatewayClassName: istio
+  listeners:
+  - name: default
+    hostname: "bar"
+    port: 80
+    protocol: HTTP
+---
+apiVersion: gateway.networking.k8s.io/v1beta1
+kind: HTTPRoute
+metadata:
+  name: http
+spec:
+  parentRefs:
+  - name: gateway
+  rules:
+  - backendRefs:
+    - name: b
+      port: 80
+`).ApplyOrFail(t)
+	apps.B[0].CallOrFail(t, echo.CallOptions{
+		Port:   echo.Port{ServicePort: 80},
+		Scheme: scheme.HTTP,
+		HTTP: echo.HTTP{
+			Headers: headers.New().WithHost("bar").Build(),
```

Comment:
Can we also add a test for unreachable when it is bar.namespace.svc.cluster.local